src/redisearch_rs/c_entrypoint/varint/src/field_mask.rs

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2006-Present, Redis Ltd.
+ * All rights reserved.
+ *
+ * Licensed under your choice of the Redis Source Available License 2.0
+ * (RSALv2); or (b) the Server Side Public License v1 (SSPLv1); or (c) the
+ * GNU Affero General Public License v3 (AGPLv3).
+*/
+
+use buffer::{BufferReader, BufferWriter};
+use encode_decode::varint::{read_field_mask, write_field_mask};
+use std::ptr::NonNull;
+
+pub type FieldMask = ffi::t_fieldMask;
+
+/// Read a varint-encoded field mask from the given buffer.
+///
+/// # Panics
+///
+/// Panics if the buffer doesn't contain a valid varint-encoded field mask.
+///
+/// # Safety
+/// The following invariants must be upheld when calling this function:
+/// 1. `b` must point to a valid `BufferReader` instance and cannot be NULL.
+/// 2. The caller must have exclusive access to the buffer reader.
+#[unsafe(no_mangle)]
+// `improper_ctypes_definitions` would be triggered because of u128 crossing the FFI boundary but
+// that's no longer an issue:
+// https://blog.rust-lang.org/2024/03/30/i128-layout-update/#compatibility
+#[allow(improper_ctypes_definitions)]
+pub extern "C" fn ReadVarintFieldMask(b: Option<NonNull<BufferReader>>) -> FieldMask {
+    let mut buffer_reader = b.unwrap();
```

Comment:
Hmm, `unwrap()`? Why do we need  an `Option` if we are just unwrapping it?

Reply:
If you accept a `NonNull<BufferReader>`, there will be no runtime check on nullability (contrary to what one might expect!).
By accepting an `Option`, we explicitly check that the pointer is not null before using it.

Reply:
Okay, fair. I thought I might be missing something